cpp/src/arrow/flight/sql/odbc/odbc_api.cc

```diff
@@ -488,7 +525,8 @@ SQLRETURN SQLGetDiagRecW(SQLSMALLINT handleType, SQLHANDLE handle, SQLSMALLINT r
     }
 
     case SQL_HANDLE_STMT: {
-      return SQL_ERROR;
+      ODBCStatement* statement = reinterpret_cast<ODBCStatement*>(handle);
+      diagnostics = &statement->GetDiagnostics();
```

Comment:
```suggestion
      auto* statement = ODBCStatement::of(handle);
      diagnostics = &statement->GetDiagnostics();
```
nit - code style; let's be consistent with rest of the SQLGetDiagRecW API for this